### PR TITLE
If repo is enabled by one user, show it to others

### PR DESCRIPTION
**Pull request by HelenaEksler**
#325

### CODE REVIEW

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php [hunk 1]
```diff
@@ -80,4 +80,48 @@ public function patchEntity($entity_id) {
...
+   * {@inheritdoc}
+   */
+  protected function checkEntityAccess($op, $entity_type, $entity) {
+    $account = $this->getAccount();
+
+    $wrapper = entity_metadata_wrapper('node', $entity);
+    $repo_name = $wrapper->og_repo->label();
+
+    if (og_is_member('node', $wrapper->og_repo->getIdentifier(), 'user', $account)) {
+      // User is member of the repository in shoov - he has access to build even
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php]: he => they ,as "he" might be a "she" ;)

**Comment by amitaibu** [comment]
Ready for review?

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php [hunk 2]
```diff
@@ -80,4 +80,48 @@ public function patchEntity($entity_id) {
...
+        'entity_type' => 'user',
+        'entity' => $account,
+        'field_name' => 'og_user_node'
+      );
+
+      og_group('node', $wrapper->og_repo->getIdentifier(), $params);
+      return TRUE;
+    }
+
+    return FALSE;
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php:120]: if user is doesn't have access we should also `og_ungroup()` them.

Comment (HelenaEksler) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php:120]: But if someone wants user to have access to the build even if user doesn't have access in github?

Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php:120]: Then it won't be possible. GitHub should be responsible for the sync.

**Comment by amitaibu** [comment]
@HelenaEksler ready for review?

**Comment by HelenaEksler** [comment]
@amitaibu Yes, ready.

File: shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php [hunk 3]
```diff
@@ -80,4 +80,43 @@ public function patchEntity($entity_id) {
...
+      'method' => 'GET',
+      'headers' => array(
+        'Authorization' => 'token ' . $access_token,
+      ),
+    );
+    $url = 'repos/' . $repo_name . '/collaborators/' . $user_name;
+    $response = shoov_github_http_request($url, $options);
+
+    if ($response['meta']['status'] == 204) {
+      // User is a member of the repository. Subscribe him to the repository.
```
Comment (amitaibu) [inline shoov/modules/custom/shoov_restful/plugins/restful/node/ci_builds/1.0/ShoovCiBuildsResource.class.php]: him => them (user can be a girl you know :wink: )

**Pull request merged**